packages/x-date-pickers/src/PickersDay/PickersDay.tsx

```diff
@@ -90,9 +91,11 @@ const useUtilityClasses = (ownerState: PickersDayProps<any>) => {
     root: [
       'root',
       selected && 'selected',
+      disabled && 'disabled',
       !disableMargin && 'dayWithMargin',
       !disableHighlightToday && today && 'today',
       outsideCurrentMonth && showDaysOutsideCurrentMonth && 'dayOutsideMonth',
+      outsideCurrentMonth && !showDaysOutsideCurrentMonth && 'hiddenDaySpacingFiller',
```

Comment:
Was not possible overriding `hiddenDaySpacingFiller` slot as this class name was not being applied